jax/random.py

```diff
@@ -988,18 +988,19 @@ def _gamma_grad(sample, a):
     grads = vmap(lax.random_gamma_grad)(alphas, samples)
   return grads.reshape(np.shape(a))
 
-def _gamma_impl(key, a):
+def _gamma_impl(key, a, use_vmap=False):
   a_shape = jnp.shape(a)
   # split key to match the shape of a
   key_ndim = jnp.ndim(key) - 1
   key = jnp.reshape(key, (-1, 2))
   key = vmap(split, in_axes=(0, None))(key, prod(a_shape[key_ndim:]))
   keys = jnp.reshape(key, (-1, 2))
   alphas = jnp.reshape(a, -1)
-  if xla_bridge.get_backend().platform == 'cpu':
```

Comment:
I do not plan to merge this PR until I understand better the reason why we don't use vmap on CPU. Presumably it does not help to use vmap over map, but does it hurt?

Reply:
@hawkinsp I would appreciate your insight in this.

Reply:
My guess is that on CPU the loop has good performance (and saves a bit of memory) but it's much slower on GPU.

This might be a nice usecase for a "gmap"

Reply:
@shoyer That must be it: the memory saving.